frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -1112,6 +1145,24 @@ run_test('begins_typeahead', () => {
     assert_typeahead_equals("test @o", all_mentions);
     assert_typeahead_equals("test @z", all_mentions);
 
+    assert_typeahead_equals("_@", false);
+    assert_typeahead_equals(" _@", false);
+    assert_typeahead_equals("test _@**o", all_mentions);
+    assert_typeahead_equals("test _@*k", all_mentions);
+    assert_typeahead_equals("test _@*h", all_mentions);
+    assert_typeahead_equals("test _@", false);
+    assert_typeahead_equals("test no_@k", false);
+    assert_typeahead_equals("_@ ", false);
+    assert_typeahead_equals("_@* ", false);
+    assert_typeahead_equals("_@** ", false);
+    assert_typeahead_equals("test\n_@i", all_mentions);
+    assert_typeahead_equals("test\n _@l", all_mentions);
+    assert_typeahead_equals("_@zuli", all_mentions);
+    assert_typeahead_equals("_@ zuli", false);
+    assert_typeahead_equals(" _@zuli", all_mentions);
+    assert_typeahead_equals("test _@k", all_mentions);
+    assert_typeahead_equals("test _@z", all_mentions);
```

Comment:
It's possible we want these to alternate with the previous block; that'll make it more likely that changes to the mention code update both versions?